lib/cicognara/bulk_entry_indexer.rb

```diff
@@ -1,16 +1,113 @@
 module Cicognara
+  class EntryIndexer
+    attr_reader :entry
+    def initialize(entry, index, cache, books_cache)
+      @entry = entry
+      @index = index
+      @cache = cache
+      @books_cache = books_cache
+    end
+
+    def books
+      @books ||= @entry.books.uniq(&:id)
```

Comment:
MarcRecord objects (retrieved from Book objects linked to Entry objects) do *not* have the full XML documents accessible using `#source` here (maybe this is due to `#uniq`?)